### ISSUE TITLE
Next generation cwise

### ISSUE THREAD

**Issue by mikolalysenko** [scijs/contributing#12]
Right now cwise has optimizations for component-wise operations, but it would be really cool to have a generic library for optimized tensor operations.  This would make it easier to do things like work with images or fields of data, and simplify things like map/reduce or tensor operations.

**Comment by jaspervdg** [comment]
scijs/cwise-compiler#2 is meant to make vector- or tensor-valued data easier to process (still leaves lots of other stuff though).

**Comment by mikolalysenko** [comment]
Yeah, that sat around too long.  I just merged it in and also added you as an owner on npm for those repos (ndarray, cwise, cwise-compiler and cwise-parser).  If you see other places where thing could be better, just go for it and fix it.
